File: java_src/Problem010.java
import java.math.*;


class Problem010
{
	public static void main(String[] args) {
		Integer [] simple_numb = new Integer[5000000];
		simple_numb[0] = new Integer(2);
		int arrind=0;
		long sum = (long)simple_numb[0];
		long lim = 2000000L;

		for (int i=0; i<=lim; i++)
		{
			if ((i>0) && (i%2>0))
			{
				for(int k=0; k<=arrind;k++)
				{
					if((i % simple_numb[k] == 0) || (i <= simple_numb[k]))
					{
						break;
					}
					if(k == arrind)
					{
						// System.out.println("i : " + i + " k : " + k + " arrind : " + arrind + " simple_numb "+ simple_numb[k]);
						arrind++;
						simple_numb[arrind] = new Integer(i);

						sum+=simple_numb[arrind];
						// if (i==(lim+1))
						// {
						// 	System.out.println("sum : " + sum);
						// }

						break;
					}
				}
			}
		}
		System.out.println("sum : " + sum);
		// long del = 2000000L;



		// while(true)
		// {
		// 	for(int i=0; i<=arrind; i++)
		// 	{
		// 		if( del % simple_numb[i] == 0)
		// 		{
		// 			del = del/simple_numb[i];
		// 			System.out.println("simple_numb " + simple_numb[i] + " del " + del);
		// 			break;
		// 		}
		// 	}
		// 	if (del == 1)
		// 		break;
		// }
	}
}

File: py_src/Problem010.py
class PrimeNumbers():

	def SumPrimeNumber(self, input_data):
		assert 0 < int(input_data), "Вводимые данные должны быть числами больше 0 числами и целыми!!!"
		simple_numb = [2]
		arrind = 0
		sum_ = 0 
		lim = int(input_data)
		for i in range(lim+1):
			if ((i>0) and (i%2>0)):
				# for k in range(arrind+1):
				for k in range(len(simple_numb)):
					#if((i % simple_numb[k] == 0) or (i <= simple_numb[k])):
					if (i <= simple_numb[k]):
						# print("failure")
						break
					if(i % simple_numb[k] == 0):
						break
					elif(k == len(simple_numb)-1):
					# if(k == arrind):
						# print("--------------------------")
						# print(i)
						# print(simple_numb[k])
						# print(i % simple_numb[k])
						simple_numb.append(i)
						#sum_ += simple_numb[arrind]
						# arrind += 1
						break
		for x in simple_numb:
			sum_ += x
		print(simple_numb)
		print( "sum : " + str(sum_))
		return sum_	

if __name__ == "__main__":
	n = PrimeNumbers()
	print(n.SumPrimeNumber(10))